pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -845,6 +845,17 @@ func (s *DiscoveryServer) startPush(version string, push *model.PushContext, ful
 		client := pending[0]
 		pending = pending[1:]
 
+		// indicates whether to do a full push for the proxy
+		proxyFull := full
+		if !full {
+			s.proxyUpdatesMutex.Lock()
+			if _, ok := s.proxyUpdates[client.modelNode.IPAddresses[0]]; ok {
+				proxyFull = true
```

Comment:
@rshriram take a look at here, if the `proxyUpdates` contains the proxy ip address, then trigger full push no matter the original push type(full/eds only) and remove it from the map.